testing/xfpga/test_metadata_c.cpp

```diff
@@ -321,4 +327,61 @@ TEST_P(metadata_c, get_bitstream_json_len) {
   EXPECT_EQ(len, -1);
 }
 
+/**
+* @test    get_interface_id_01
+* @brief   Tests: get_interface_id
+* @details Given invalid params, the function returns FPGA_INVALID_PARAM
+*/
+TEST_P(metadata_c, get_interface_id_01) {
+  uint64_t id_l;
+  uint64_t id_h;
+  auto _token = (struct _fpga_token *)tokens_[0];
+  ASSERT_EQ(FPGA_OK, xfpga_fpgaOpen(tokens_[0], &handle_, 0));
+
+  // Invalid object type
+  _token->magic = 0x123;
+  auto res = get_interface_id(handle_, &id_l, &id_h);
+  EXPECT_EQ(res, FPGA_INVALID_PARAM);
+
+  _token->magic = FPGA_TOKEN_MAGIC;
+
+  res = get_interface_id(handle_, nullptr, nullptr);
+  EXPECT_EQ(res, FPGA_INVALID_PARAM);
+}
+
+/**
+* @test    get_interface_id_02
+* @brief   Tests: get_interface_id
+* @details Given invalid params, the function returns FPGA_INVALID_PARAM
+*/
+TEST_P(metadata_c, get_interface_id_02) {
+  uint64_t id_l;
+  uint64_t id_h;
+  ASSERT_EQ(FPGA_OK, xfpga_fpgaOpen(tokens_[0], &handle_, 0));
+  struct _fpga_handle  *handle = (struct _fpga_handle *)handle_;
+
+  handle->token = NULL;
+  auto res = get_interface_id(handle_, &id_l, &id_h);
+  EXPECT_EQ(res, FPGA_INVALID_PARAM);
+}
+
+/**
+* @test    get_interface_id_03
+* @brief   Tests: get_interface_id
+* @details Given invalid params, the function returns FPGA_INVALID_PARAM
```

Comment:
get_interface_id apparently returns FPGA_EXCEPTION in this case, not FPGA_INVALID_PARAM.